iop/fs/bdm/src/part_driver.c

```diff
@@ -83,8 +87,25 @@ static int part_getPartitionTable(struct block_device* bd, part_table* part)
     if (sbuf[0x1FE] == 0x55 && sbuf[0x1FF] == 0xAA) {
         for (i = 0; i < 4; i++) {
             part_raw = (part_raw_record*)(sbuf + 0x01BE + (i * 16));
-            if (part_getPartitionRecord(bd, part_raw, &part->record[i]) != 0)
-                return 0; //Invalid record encountered, so the table is probably invalid.
```

Comment:
@rickgaiser I am not sure but should there be == 0 ? I guess that you meant that partition fails if the function returns zero, but it looks like you did it opposite. Can it be a typo? Or maybe I misunderstood something.

Reply:
> If `bd->sectorCount` is zero

huh, I mean that `bd->sectorCount` isnt initialized anywhere. What is assigned there before partition manager starts probing everything?

Reply:
The block device drivers create first block device (p0), being the entire drive:

https://github.com/ps2dev/ps2sdk/blob/898cefeb3b2d7cc56bd59a4a92c3cbbd37dd21cd/iop/usb/usbmass_bd/src/scsi.c#L168

https://github.com/ps2dev/ps2sdk/blob/898cefeb3b2d7cc56bd59a4a92c3cbbd37dd21cd/iop/iLink/IEEE1394_bd/src/scsi.c#L168

https://github.com/ps2dev/ps2sdk/blob/898cefeb3b2d7cc56bd59a4a92c3cbbd37dd21cd/iop/sio/mx4sio_bd/src/mx4sio.c#L845

The partition drivers then create new block devices on top of the entire drive block device, like p1, p2, p3.

Reply:
Oh, thanks!